diff --git a/llama/llama.cpp/.rsync-filter b/llama/llama.cpp/.rsync-filter
index 7987be1c..88e475ad 100644
--- a/llama/llama.cpp/.rsync-filter
+++ b/llama/llama.cpp/.rsync-filter
@@ -8,14 +8,18 @@ include /common/json.*
 include /common/log.*
 include /common/unicode.*
 include /common/sampling.*
+include /common/reasoning-budget.*
 include /include/
 include /include/llama.*
 include /include/llama-*.*
 include /tools/
 include /tools/mtmd/
 include /tools/mtmd/*.h
+include /tools/mtmd/debug/
+include /tools/mtmd/debug/*
 include /tools/mtmd/clip.cpp
 include /tools/mtmd/mtmd.cpp
 include /tools/mtmd/mtmd-audio.cpp
+include /tools/mtmd/mtmd-image.cpp
 include /tools/mtmd/mtmd-helper.cpp
 include /tools/mtmd/models/
 include /tools/mtmd/models/*.h
